### PR TITLE
ewma - initial commit

### PR DESCRIPTION
**Pull request by rjenc29**
Just some idle thoughts whilst watching James Bond - not yet ready to properly submit for PR onto the main repo.  But check out the notebook / speedup when jit compiled...

### CODE REVIEW

**Comment by rjenc29** [comment]
more or less a straight port of pandas/_libs/window.pyx

File: tests/utilities/test_ewma.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import pandas as pd
+import numpy as np
+
+from fastats.utilities.ewma import ewma
+
+
+def sample_data_series():
```
Comment (dwillmer) [inline tests/utilities/test_ewma.py:7]: nice! would be good to parameterise this, as in `sample_data_series(n, dtype='float')`, or use hypothesis\reasonable to generate the data

**Comment by dwillmer** [comment]
Yea, very nice!

The speedup is really impressive, definitely worthwhile, but in my head the API would be something like
```python
result = single_pass(data, value=ewma, window=22)
```
or
```python
result = windowed_pass(data, value=mean, weighting=exp, size=22)
```

ie, we want to separate the iteration over the data from the calculation of the result.

I've been focussing on some edge cases with the numba jit recently, but sorting rolling window APIs is definitely the next big item to add.

Cheers!

**Comment by rjenc29** [comment]
Closing this ancient PR - to be included in due course

**Pull request closed without merging**